### PR TITLE
add default to Get transformation

### PR DESCRIPTION
**Pull request by erikdies**
https://percolate.atlassian.net/browse/SCRM-824

This adds a `default` option to `Get` in transformations.

### CODE REVIEW

**Comment by lapointexavier** [comment]
Makes sense to me. @ltucker, thoughts? 

:shipit:

File: bfh/transformations.py [hunk 1]
```diff
@@ -111,18 +111,19 @@ def __init__(self, *args, **kwargs):
...
     def _get_from_dict(self, source, path):
         if not self.required:
-            return source.get(path)
+            return source.get(path) or self.default
         return source[path]
 
     def _get_from_obj(self, source, path):
         if not self.required:
-            return getattr(source, path, None)
+            return getattr(source, path, None) or self.default
```
Comment (lapointexavier) [inline bfh/transformations.py:126]: Could you do : 
```
getattr(source, path, self.default)
```

?

File: bfh/transformations.py [hunk 2]
```diff
@@ -111,18 +111,19 @@ def __init__(self, *args, **kwargs):
...
         self.required = kwargs.get('required', False)
+        self.default = kwargs.get('default')
 
     def __call__(self, source):
         return self.function(source)
 
     def _get_from_dict(self, source, path):
         if not self.required:
-            return source.get(path)
+            return source.get(path) or self.default
```
Comment (lapointexavier) [inline bfh/transformations.py:121]: `source.get(path, self.default)`
Same here?

File: bfh/transformations.py (hunk 1 above)
Comment (erikdies) [reply bfh/transformations.py:126]: I initially had it that way, but I can't - there is an issue with nested paths, e.g. `Get("outer", "middle", "inner")`.

File: bfh/transformations.py (hunk 2 above)
Comment (erikdies) [reply bfh/transformations.py:121]: Same as above

**Comment by benauthor** [comment]
You should update the docs to include this new feature. see `doc` target in makefile.

**Comment by lapointexavier** [comment]
Also release it on Pypi?

**Comment by benauthor** [comment]
indeed

**Pull request merged**